Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -73,18 +75,63 @@ private void OnEmagged(EntityUid uid, AccessReaderComponent reader, ref GotEmagg
             Dirty(reader);
         }
 
+        /// <summary>
+        /// Finds all AccessReaderComponents in the container of the
+        /// required entity.
+        /// </summary>
+        /// <param name="target">The entity to search for a container</param>
+        private List<AccessReaderComponent> FindAccessReadersInContainer(
```

Comment:
Thinking about it, couldnt you just move the loop, check and stuff to `FindAccessItemsInventory()` with a comment like `// maybe its inside a container` and leave the IsAllowed below alone

Reply:
`FindAccessItemsInventory` finds all AccessComponents, `FindAccessReadersInContainer` finds AccessReaderComponents. I don't think it is good idea to mix all entities with Access and AccessReader components in one HashSet. Or I don't quite understand how it should work.

Reply:
Ah, you are right, i forgot the access reader system handled some of acesscomponent as well for some reason.